hail/src/test/scala/is/hail/fs/FSSuite.scala

```diff
@@ -271,6 +271,60 @@ trait FSSuite {
     assert(!fs.exists(f))
   }
 
+  @Test def testReadWriteManyBytes(): Unit = {
+    val f = t()
+
+    using(fs.create(f)) { os =>
+      Array.tabulate(1000000)(i => os.write(i))
+    }
+
+    assert(fs.exists(f))
+
+    using(fs.open(f)) { is =>
+      Array.tabulate(1000000)(i => assert(i == is.read()))
+    }
+
+    fs.delete(f, false)
+
+    assert(!fs.exists(f))
+  }
+
+  @Test def testReadWriteBytesLargerThanBuffer(): Unit = {
+    val f = t()
+
+    val byteBufferSize = 64 * 1024
```

Comment:
This test is a bit brittle because it depends on an exact value of an implementation detail of the FS. If we were to change the implementation, the test could become invalid, and we might forget to update the tests. I think it is unlikely that we would ever use a buffer size greater than, say ten megabytes, so let's instead ensure that our tests cover a spread of different sizes that are likely to catch this edge case regardless of the exact buffer size.